contracts/ssnlist.scilla

```diff
@@ -1542,3 +1575,485 @@ transition DrainContractBalance(amt: Uint128, initiator: ByStr20)
+  forall ssninfo_list SwapBuffDeposit;
+  forall ssninfo_list SwapDirectDeposit;
+  forall ssninfo_list SwapDelegStakePerCycle;
+  forall ssninfo_list SwapDepositAmtDeleg;
+  forall ssninfo_list SwapSSNDelegAmt;
+  forall ssninfo_list SwapLastBuffDepositCycleDeleg;
+  forall ssninfo_list SwapLastWithdrawCycleDeleg;
+  
+  (* swap pending withdrawal *)
+  match c with
+  | Some initial_deleg =>
+    deleg_blk_map <- withdrawal_pending[initial_deleg];
+    match deleg_blk_map with
+    | Some deleg_blk_entry =>
+      cycle_reward_list = builtin to_list deleg_blk_entry;
+      forall cycle_reward_list SwapPendingWithdrawal;
+      delete withdrawal_pending[initial_deleg]
+    | None => (* no pending withdrawal *)
+    end
+  | None =>
+    e = DelegDoesNotExistAtSSN;
+    ThrowError e
+  end
+end
+
+
+(* @dev: Creates a request to another delegator to indicate interest in transferring all stakes, rewards etc to the delegator *)
+(* @param initiator: Address of the requestor; the one who call the proxy *)
+(* @param new_deleg_addr: Address of the new deleg that will be receiving all the stakes from requestor *)
+(* requestor can override the new deleg address by sending a new request *)
+transition RequestDelegatorSwap(initiator: ByStr20, new_deleg_addr: ByStr20)
+  IsNotPaused;
+  IsProxy;
+  deleg_swap_request[initiator] := new_deleg_addr;
+  e = { _eventname: "RequestDelegatorSwap"; initial_deleg: initiator; new_deleg: new_deleg_addr };
+  event e
+end
+
+(* @dev: Accepts a swap request from a requestor. *)
+(* Once confirmed, everything which was under the requestor would be transferred over to the new deleg *)
+(* @param initiator: Address of the new_deleg confirming the request *)
+(* @param requestor: Address of the one that creates the swap request *)
+(* Transfer cannot be reverted once confirmed *)
+transition ConfirmDelegatorSwap(initiator: ByStr20, requestor: ByStr20)
+  IsNotPaused;
+  IsProxy;
+  swap_addr <- deleg_swap_request[requestor];
+  match swap_addr with
+  | Some requestor_swap_addr =>
+    is_new_deleg_valid = builtin eq requestor_swap_addr initiator;
+    match is_new_deleg_valid with
+    | True =>
+      SwapDelegator requestor requestor_swap_addr;
+      delete deleg_swap_request[requestor];
+      e = { _eventname: "ConfirmDelegatorSwap"; initial_deleg: requestor; new_deleg: initiator };
+      event e
+    | False =>
+      (* throw error - requestor swap address is different from this initiator *)
+      e = SwapAddrValidationFailed;
+      ThrowError e
+    end
+  | None => (* throw error *)
+    e = InvalidSwapAddr;
+    ThrowError e
+  end
+end
+
+(* @dev: Cancels a swap request. *)
+(* @param initiator: Address of requestor who wants to cancel the swap process *)
+transition CancelDelegatorSwap(initiator: ByStr20)
+  IsNotPaused;
+  IsProxy;
+  swap_addr <- deleg_swap_request[initiator];
+  match swap_addr with
+  | Some requestor_swap_addr =>
+    delete deleg_swap_request[initiator];
```

Comment:
will this have the empty map issue?

Reply:
this line won't have the issue because `deleg_swap_request` is a single layer map but there are other parts of the swap process which involves the same maps as the "delete pr". Hence, there will still be the empty map issue in this pr.